packages/cadl-ranch-specs/http/payload/json-merge-patch/main.tsp

```diff
@@ -0,0 +1,222 @@
+import "@typespec/http";
+import "@azure-tools/cadl-ranch-expect";
+
+using TypeSpec.Http;
+
+@doc("Test for merge-patch+json content-type")
+@scenarioService("/json-merge-patch")
+namespace Payload.JsonMergePatch;
+
+@doc("Details about a resource.")
+model Resource {
+  name: string;
+  description?: string;
+  map?: Record<InnerModel>;
+  array?: InnerModel[];
+  intValue?: int32;
+  floatValue?: float32;
+}
+
+@doc("Details about a resource for patch operation.")
+model ResourcePatch {
+  description?: string;
+  map?: Record<InnerModel>;
+  array?: InnerModel[];
```

Comment:
It may be `InnerModelPatch`?

Reply:
and seems it lacks a testing point that a property is a model?

Reply:
@pshao25 you are asking that every model that referred by a PATCH op have all properties optional.

In reality this probably not going to happen. It is EVERY models referred by the PATCH op.

In the cadl-ranch, I think Haoling can make the change to have the `name` there optional as well. But in reality, you will need a lint to prevent anyone doing this (provided that we all agree that required property should not appear).

Reply:
and I think we need an array of primitive type.

Reply:
Currently I'm adding cases that service uses in priority, i.e. Record<InnerModel> and string. Also add some typical cases.

Generally if you covered `InnerModel[]` and `Record<InnerModel>`, `InnerModel` property should be covered fine. Typespec does not have primitive types, Java/.Net have, but if it's optional property, it's not primitive type as well. 

If .Net wants to have these two test cases specifically, I can add in this pr. But in the future, I think we can increamentally add test cases to json-merge-patch scenario when we encounter more use cases.